### ISSUE TITLE
Regent: Inliner bug in FFT

### ISSUE THREAD

**Issue by elliottslaughter** [StanfordLegion/legion#946]
This is an inliner bug triggered by the Regent FFT implementation.

```
git clone -b 356581f46bf49716c96b7d6b0249a44719c6605b https://github.com/StanfordLegion/regent-fft.git
cd regent-fft
./install.py
source env.sh
../regent.py test/fft_test.rg
```

You'll see this passes.

Now uncomment this line: https://github.com/StanfordLegion/regent-fft/blob/356581f46bf49716c96b7d6b0249a44719c6605b/src/fft.rg#L144

You should see a failure like:

```
.../src/regent/inline_tasks.t:305: assertion failed! 
stack traceback: 
        [C]: in function 'assert' 
        .../src/regent/inline_tasks.t:305: in function 'expr' 
        .../src/regent/inline_tasks.t:347: in function <.../src/regent/inline_tasks.t:345> 
        .../src/regent/inline_tasks.t:370: in function 'stat' 
        .../src/regent/inline_tasks.t:374: in function 'fn' 
        src/terralist.lua:150: in function 'map' 
        .../src/regent/inline_tasks.t:374: in function 'block' 
        .../src/regent/inline_tasks.t:323: in function <.../src/regent/inline_tasks.t:322> 
        .../src/regent/inline_tasks.t:370: in function 'stat' 
        .../src/regent/inline_tasks.t:374: in function 'fn' 
        ... 
        .../src/regent/inline_tasks.t:537: in function <.../src/regent/inline_tasks.t:533> 
        .../src/regent/inline_tasks.t:587: in function 'stat' 
        .../src/regent/inline_tasks.t:592: in function 'fn' 
        src/terralist.lua:150: in function 'map' 
        .../src/regent/inline_tasks.t:592: in function 'block' 
        .../src/regent/inline_tasks.t:597: in function 'top_task' 
        .../src/regent/inline_tasks.t:607: in function <.../src/regent/inline_tasks.t:601> 
        ...regent/src/regent/passes.t:63: in function <...regent/src/regent/passes.t:57> 
        ...gent/regent-fft/src/fft.rg:160: in function 'generate_fft_interface' 
        test/fft_test.rg:19: in main chunk
```

**Comment by magnatelee** [comment]
Should be fixed by f9800ab1. @elliottslaughter Can you pull and try again?

**Comment by elliottslaughter** [comment]
Confirmed. Thanks!
